src/main/java/com/keniu/services/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,80 @@
+package com.keniu.services;
+
+import com.keniu.dto.CreateCartItemRequestDto;
+import com.keniu.dto.ShoppingCartDto;
+import com.keniu.dto.UpdateCarItemRequestDto;
+import com.keniu.exceptions.EntityNotFoundException;
+import com.keniu.mappers.ShoppingCartMapper;
+import com.keniu.models.Book;
+import com.keniu.models.CartItem;
+import com.keniu.models.ShoppingCart;
+import com.keniu.repositories.ShoppingCartRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
```

Comment:
Please, add @Transactional above all methods that save/update/delete data from the DB to make it more obvious

Reply:
Ok, added